### PR TITLE
New methods for collections

### PR DESCRIPTION
**Pull request by caalador**
Enable add and remove for
collection.

part of #8497

### CODE REVIEW

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 1]
```diff
@@ -77,23 +79,58 @@
...
+     * @return this ListDataView instance
+     * @throws UnsupportedOperationException
+     *         if backing collection doesn't support modification
+     * @throws IllegalArgumentException
+     *         if item doesn't exist or collection is not a list
+     */
+    V addItemBefore(T item, T before);
+
+    /**
+     * Add multiple items to the data list.
```
Comment (mshabarov) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:116]: Add -> Adds

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java [hunk 2]
```diff
@@ -77,23 +79,58 @@
...
+     *
+     * @param items
+     *         collection of item to add
+     * @return this ListDataView instance
+     * @throws UnsupportedOperationException
+     *         if backing collection doesn't support modification
+     */
```
Comment (mshabarov) [inline flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:123]: Add a link to contiguous methods via `@see` (addItems <-> removeItems, addItemsBefore <-> addItemsAfter)

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 1 above)
Comment (caalador) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java]: done

File: flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java (hunk 2 above)
Comment (caalador) [reply flow-data/src/main/java/com/vaadin/flow/data/provider/ListDataView.java:134]: done

**Pull request merged**